clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryClient.java

```diff
@@ -124,35 +58,18 @@ public void execute(DataRecoveryParams drParams, StoreRepushCommand.Params cmdPa
       return;
     }
 
-    Map<String, Pair<Boolean, String>> pushMap = getRepushViability(storeNames, cmdParams);
-    Set<String> filteredStoreNames = new HashSet<>();
-
-    for (Map.Entry<String, Pair<Boolean, String>> e: pushMap.entrySet()) {
-      if (e.getValue().getLeft()) {
-        filteredStoreNames.add(e.getKey());
-      } else {
-        this.getExecutor().getSkippedStores().add(e.getKey());
-      }
+    if (!drParams.isNonInteractive && !confirmStores(storeNames)) {
+      return;
     }
+    getExecutor().perform(storeNames, cmdParams);
 
-    if (!filteredStoreNames.isEmpty()) {
-      if (!drParams.isNonInteractive && !confirmStores(filteredStoreNames)) {
-        return;
+    for (DataRecoveryTask t: getExecutor().getTasks()) {
+      StoreRepushCommand cmd = (StoreRepushCommand) t.getCommand();
+      if (cmd.getResult().isError()) {
```

Comment:
This condition seems not quite right, `isError()` would also include the ones that have been executed, but ended with some sort of error, so it does not always equal to the ones got skipped. This makes me to think again about the benefit of introducing the enum for storing the result of `getRepushViability()` in `StoreRepushCommand` and then filtered it here (or some other approaches as well)

Reply:
Introduced a class to hold repush viability info in a more organized way.